### ISSUE TITLE
split_pairs not working

### ISSUE THREAD

**Issue by femiliani** [nanoporetech/duplex-tools#34]
Really excited to try out the new split_pairs on some dorado basecalled data but i cant seem to get it to work. 

I managed to use pair without problems. but split pair gives the following error (running on 4000 reads for speed). paths shortened for clarity:
```
(duplex) (base) [francesco testing]$ duplex_tools split_pairs dorado_singlex.bam pod5_test/ split_duplex/
[12:30:15 - SplitPairs] Find split locations.
[12:30:15 - SplitPairs] Self-align reads from: dorado_singlex.bam
Traceback (most recent call last):
  File "/ont-dependencies/duplex_tools/venv/bin/duplex_tools", line 33, in <module>
    sys.exit(load_entry_point('duplex-tools==0.2.20', 'console_scripts', 'duplex_tools')())
  File "/ont-dependencies/duplex_tools/venv/lib/python3.8/site-packages/duplex_tools/__init__.py", line 39, in main
    args.func(args)
  File "/ont-dependencies/duplex_tools/venv/lib/python3.8/site-packages/duplex_tools/split_pairs.py", line 134, in main
    split_pairs(
  File "/ont-dependencies/duplex_tools/venv/lib/python3.8/site-packages/duplex_tools/split_pairs.py", line 40, in split_pairs
    split_locations = get_split_points(
  File "/ont-dependencies/duplex_tools/venv/lib/python3.8/site-packages/duplex_tools/split_pairs_steps.py", line 63, in get_split_points
    chunk3 = [next(it) for _ in range(4000)]
  File "/ont-dependencies/duplex_tools/venv/lib/python3.8/site-packages/duplex_tools/split_pairs_steps.py", line 63, in <listcomp>
    chunk3 = [next(it) for _ in range(4000)]
  File "pysam/libcalignmentfile.pyx", line 2209, in pysam.libcalignmentfile.IteratorRowAll.__next__
StopIteration
```

Any suggestions for ways to fix this? 

Thanks so much! 

-Francesco

**Comment by ollenordesjo** [comment]
Hi @femiliani! yes, I noticed this today as well actually. The underlying reason is that you have less than 5000 total reads in the dataset.

The way to fix it is to skip the initial chunk-wise iterations through the first 5000 reads:

```
           chunk1 = [next(it) for _ in range(400)]
            chunk2 = [next(it) for _ in range(600)]
            chunk3 = [next(it) for _ in range(4000)]
            iterator = chain([chunk1, chunk2, chunk3],
                             chunked(it, chunk_size))
```

should just become:
```
            iterator = chunked(it, chunk_size)
```

I will get this fixed in next version. Thanks for noticing!

**Comment by femiliani** [comment]
Thank you, @ollenordesjo! Seems like my need for speed only slowed me down. Thanks so much for the help!

**Comment by ollenordesjo** [comment]
Hi @femiliani, this should now be fixed with 0.3.0.
